ubyssey/mixins.py

```diff
@@ -0,0 +1,276 @@
+import datetime
+from datetime import datetime
+
+from django.utils import timezone
+import pytz
+from random import randint, choice
+
+from django.conf import settings
+from django.http import Http404
+from django.db import connection
+from django.db.models import Case, ExpressionWrapper, DurationField, F, FloatField, OuterRef, Subquery, Value, When 
+from django.db.models.aggregates import Count
+
+from dispatch.models import Article, Page, Section, Subsection, Podcast, Image, ImageAttachment
+
+from ubyssey.events.models import Event
+
+
+class ArticleMixin(object):
```

Comment:
Great job consolidating all of the article-related logic. A few points:

- Did you consider using dependency injection rather than a mixin? Mixins are a powerful tool for inheritance, but in this case it's just being used to provide additional functionality to a class. Having this functionality be implicitly mixed in with the the view logic (e.g. in `UbysseyHomePageView`) feels like it could get a bit confusing.
- I think it might be worthwhile to split this class up even further in the future. Some methods use the database and some don't, but overall it's quite a bit of logic that might benefit from being moved into several smaller classes.

Reply:
I defaulted to mixins, though I did hesitate, because it's so common to see code that has to be reused with View classes in particular take the form of mixins. It at least seemed a step up from having this code in a helper class. I agree it still needs work.

Wasn't sure about dependency injections because I've found Django's kind of opaque about the lifetime of the View object. I don't know when the service object would have to be instantiated.

Definitely agreed about splitting it up further. It will probably retain some stuff but other stuff it was lumped with originally (not my doing) seems like it really ought to be abstracted into its own thing.